### PR TITLE
Prevent navigation out of initial origin

### PR DESCRIPTION
**Pull request by n-riesco**
Fixes plotly/streambed#10963

Closes #433

### CODE REVIEW

**Comment by n-riesco** [comment]
@tarzzz In the end, it turned out easier to listen on `will-navigate`. Would you do the review, please?

---

I don't think Falcon needs `nodeIntegration` to be enabled, so I've opened issue #438 to remind us to disable it.

File: backend/main.development.js [hunk 1]
```diff
@@ -59,6 +59,12 @@ app.on('ready', () => {
         mainWindow.openDevTools();
     }
 
+    // prevent navigation out of HTTP_URL
+    // see https://electronjs.org/docs/api/web-contents#event-will-navigate
+    mainWindow.webContents.on('will-navigate', (event, url) => {
+        if (!url.startsWith(HTTP_URL)) event.preventDefault();
+    });
+
```
Comment (tarzzz) [inline backend/main.development.js]: 👍

**Comment by tarzzz** [comment]
btw, You can cherry-pick https://github.com/plotly/falcon-sql-client/pull/433/commits/bd1df53571f4168fc56ef38c5d6b567b733cf014 here to close #438   as well.. !!

**Comment by n-riesco** [comment]
@tarzzz 
- the reason I opened #438 is that I don't want `nodeIntegration` in this PR.
- I can't test the on-prem case, please, could you test this PR doesn't break oauth?

**Comment by tarzzz** [comment]
Ok, I will check onprem (Not today, probably monday) ..

**Comment by n-riesco** [comment]
@tarzzz I've just realised that we can't test this PR against on-prem, because oauth in the desktop app is still broken (#306).

I've tested it against Plotly Cloud and it's working:

peek 2018-05-19 07-45

**Comment by tarzzz** [comment]
Yeah, we should be good to merge it because the issue is only applicable on the electron app. drag-and-drop on browser version will simply load the new url in that tab.

**Pull request merged**